packages/react-components/src/components/RemoteVideoTile.tsx

```diff
@@ -99,22 +113,80 @@ export const _RemoteVideoTile = React.memo(
       );
     }, [renderElement, isReceiving]);
 
+    const showLabelTrampoline = useMemo(() => {
+      /* @conditional-compile-remove(one-to-n-calling) */
+      /* @conditional-compile-remove(PSTN-calls) */
+      return canShowLabel(participantState, props.isNarrow, props.showLabel, containerWidth);
+      return props.showLabel;
+    }, [
+      /* @conditional-compile-remove(one-to-n-calling) */
+      /* @conditional-compile-remove(PSTN-calls) */
+      participantState,
+      /* @conditional-compile-remove(one-to-n-calling) */
+      /* @conditional-compile-remove(PSTN-calls) */
+      props.isNarrow,
+      /* @conditional-compile-remove(one-to-n-calling) */
+      /* @conditional-compile-remove(PSTN-calls) */
+      containerWidth,
+      props.showLabel
+    ]);
+
     return (
-      <VideoTile
-        key={userId}
-        userId={userId}
-        renderElement={renderVideoStreamElement}
-        displayName={displayName}
-        onRenderPlaceholder={onRenderAvatar}
-        isMuted={isMuted}
-        isSpeaking={isSpeaking}
-        showMuteIndicator={showMuteIndicator}
-        showLabel={props.showLabel}
-        personaMinSize={props.personaMinSize}
-        /* @conditional-compile-remove(one-to-n-calling) */
-        /* @conditional-compile-remove(PSTN-calls) */
-        participantState={participantState}
-      />
+      <div ref={containerRef}>
+        <VideoTile
+          key={userId}
+          userId={userId}
+          renderElement={renderVideoStreamElement}
+          displayName={displayName}
+          onRenderPlaceholder={onRenderAvatar}
+          isMuted={isMuted}
+          isSpeaking={isSpeaking}
+          showMuteIndicator={showMuteIndicator}
+          personaMinSize={props.personaMinSize}
+          showLabel={showLabelTrampoline}
+          /* @conditional-compile-remove(one-to-n-calling) */
+          /* @conditional-compile-remove(PSTN-calls) */
+          participantState={participantState}
+        />
+      </div>
     );
   }
 );
+
+/* @conditional-compile-remove(one-to-n-calling) */
+/* @conditional-compile-remove(PSTN-calls) */
+/**
+ * Checks if a participant state is calling or hold.
+ * These states match the states used to render the participantStateString in VideoTile and ParticipantItem.
+ */
+const isCallingOrHold = (participantState?: ParticipantState): boolean => {
+  return !!participantState && ['Idle', 'Connecting', 'EarlyMedia', 'Ringing', 'Hold'].includes(participantState);
+};
+
+/* @conditional-compile-remove(one-to-n-calling) */
+/* @conditional-compile-remove(PSTN-calls) */
+/**
+ * Determines if a label should be shown for a remote video tile.
+ * When the remote video tile is rendered as a small tile in horizontal gallery,
+ * we hide the participants name if they are in hold/connecting states.
+ */
+const canShowLabel = (
```

Comment:
is it possible to move this logic into the VideoGallery? this is a layer up from the VideoTile component but its in a interesting spot living in the internal wrapper (_RemoteVideoTile) for the VideoTile that the VideoGallery uses.

Reply:
Actually never mind, doing this here is very clever since Contoso has the ability to override this in the VideoGallery level.

Reply:
Plus video gallery is already doing too much